pinot-common/src/main/java/org/apache/pinot/common/exception/QueryException.java

```diff
@@ -180,6 +185,20 @@ public static ProcessingException getException(ProcessingException processingExc
     return copiedProcessingException;
   }
 
+  public static void decorateException(ProcessingException processingException) {
+    String errorType;
+    switch (processingException.getErrorCode()) {
+      case SERVER_INSTANCE_MISSING_ERROR_CODE:
+        errorType = "ServerInstanceMissingError";
+        break;
+      // TODO: add more branches
```

Comment:
I don't like this idea (used above in `getException`) where we have a base exception with the message and then we do a deep copy. IMHO it would be better to just have an exception class for each ProcessingException or a builder method that takes the error code and the message and does something like this method, creating the exception with the `errorType` prefix associated with the given error code.

I'm not using this method as I would like because I'm following the existent pattern where the exception is first thrown without the `errorType` prefix and then it is decorated.

In case we want to continue using the pattern in this PR, we just need to add here the prefix for each error code (or use an String[] or a map from int to string) and throw a `ProcessingException` withouth the error type prefix